src/main/java/com/auction/controller/AuctionController.java

```diff
@@ -26,12 +26,17 @@ public List<Auction> getALlAuctions() {
     }
 
     @GetMapping("/{auctionId}")
-    public Auction getAuctionById(@PathVariable int auctionId) {
-        return auctionService.findById(auctionId);
+    public ResponseEntity<Auction> getAuctionById(@PathVariable int auctionId) {
+        Auction auction = auctionService.findById(auctionId);
+        if (auction != null) {
```

Comment:
in such case when you have only two conditions it is better to use ternary operations . 
Besides Controller should not decide which response type will be . So in your case you should to use ControllerAdvice to handle Exceptions and wrap them into ResponseEntity. So that you will have one place where you catch exceptions.